### PR TITLE
Improve test on TypeName

### PR DESCRIPTION
**Pull request by danglotb**
Hello,

I open this pull request to specify the line [197](https://github.com/square/javapoet/blob/master/src/main/java/com/squareup/javapoet/TypeName.java#L197) in the ```equals()``` method of ```com.squareup.javapoet.TypeName```.

```java
if (o == null) return false;
```

### CODE REVIEW

File: src/test/java/com/squareup/javapoet/TypeNameTest.java [hunk 1]
```diff
@@ -178,5 +179,6 @@ private void assertEqualsHashCodeAndToString(TypeName a, TypeName b) {
     assertEquals(a.toString(), b.toString());
     assertThat(a.equals(b)).isTrue();
     assertThat(a.hashCode()).isEqualTo(b.hashCode());
+    assertFalse(a.equals(null));
```
Comment (ronshapiro) [inline src/test/java/com/squareup/javapoet/TypeNameTest.java]: `assertThat(a).isNotNull();`

Comment (tbroyer) [inline src/test/java/com/squareup/javapoet/TypeNameTest.java]: He, no: this specifically about testing the `.equals(null)` contract 😉

Comment (sormuras) [inline src/test/java/com/squareup/javapoet/TypeNameTest.java]: See https://github.com/square/javapoet/blob/master/src/main/java/com/squareup/javapoet/TypeName.java#L197

Comment (ronshapiro) [inline src/test/java/com/squareup/javapoet/TypeNameTest.java]: Oh fair. What about `assertThat(a).isNotEqualTo(null)'?

Comment (sormuras) [inline src/test/java/com/squareup/javapoet/TypeNameTest.java]: Would be more inline with
```java
assertThat(a.equals(b)).isTrue();
assertThat(a.hashCode()).isEqualTo(b.hashCode());
```
I think `assertThat(a.equals(a)).isTrue();` is missing, too. Testing the [if (this == o) return true;](https://github.com/square/javapoet/blob/master/src/main/java/com/squareup/javapoet/TypeName.java#L196) contract.

Comment (JakeWharton) [inline src/test/java/com/squareup/javapoet/TypeNameTest.java]: True, but that's just an optimization. If `a.equals(b)` works then `b.equals(a)` must also work, thus we then know `a.equals(a)` will work.

Comment (sormuras) [inline src/test/java/com/squareup/javapoet/TypeNameTest.java]: True. True. Stays true. 🥂

**Pull request merged**